src/org/omegat/core/dictionaries/StarDict.java

```diff
@@ -252,16 +257,16 @@ private void addIndex(final String key, final int start, final int len, final Ma
          */
         @Override
         public List<DictionaryEntry> readArticles(String word) {
-            Object dictData = data.get(word);
-            if (dictData == null) {
-                return Collections.emptyList();
-            }
             List<DictionaryEntry> result = new ArrayList<DictionaryEntry>();
-            if (dictData instanceof Entry) {
-                result.add(new DictionaryEntry(word, ((Entry) dictData).getArticle()));
-            } else if (dictData instanceof Entry[]) {
-                for (Entry entry : (Entry[]) dictData) {
-                    result.add(new DictionaryEntry(word, entry.getArticle()));
+            Iterable<String> it = data.predictiveSearch(word);
```

Comment:
*Always* doing a predictive search is going to be very noisy. I think we need to only do this when `Preferences.DICTIONARY_FUZZY_MATCHING` is `true`.